### ISSUE TITLE
Feature Request:  Assign multiple MACs to a single lease/IP

### ISSUE THREAD

**Issue by ackstorm23** [TechnitiumSoftware/DnsServer#717]
Officially requesting a new feature to assign multiple MAC addresses to a single lease/IP address. 

Details were discussed in this discussion thread

https://github.com/TechnitiumSoftware/DnsServer/discussions/716#

**Comment by noseshimself** [comment]
No discussion in that thread, just a feature request.

Just one question: What is supposed to happen if a second device is showing up while the first has not actively returned the IP address? That's the biggest problem with this kind of setup and without any reasonable solution so far. No matter what you do, things can (and will) go wrong.

**Comment by ShreyasZare** [comment]
> Officially requesting a new feature to assign multiple MAC addresses to a single lease/IP address.
> 
> Details were discussed in this discussion thread
> 
> #716

Thanks for the feature request.

**Comment by ShreyasZare** [comment]
> No discussion in that thread, just a feature request.
> 
> Just one question: What is supposed to happen if a second device is showing up while the first has not actively returned the IP address? That's the biggest problem with this kind of setup and without any reasonable solution so far. No matter what you do, things can (and will) go wrong.

With such an advance option, if something goes wrong then its on the administrator since they knew it could have issue if not used correctly.

**Comment by ackstorm23** [comment]
> No discussion in that thread, just a feature request.

Fair.  At the time I was asking if the feature existed. 

> Just one question: What is supposed to happen if a second device is showing up while the first has not actively returned the IP address? That's the biggest problem with this kind of setup and without any reasonable solution so far. No matter what you do, things can (and will) go wrong.

As with my DHCP servers, a basic ping check to make sure the IP is not responding before it hands it out should avoid two devices being given it at the same time.

The whole idea with this feature is for devices which switch between network links (such as wired and wireless) and how the MAC address change would force them to have different IPs without this feature.

**Comment by ackstorm23** [comment]
> > No discussion in that thread, just a feature request.
> > Just one question: What is supposed to happen if a second device is showing up while the first has not actively returned the IP address? That's the biggest problem with this kind of setup and without any reasonable solution so far. No matter what you do, things can (and will) go wrong.
> 
> With such an advance option, if something goes wrong then its on the administrator since they knew it could have issue if not used correctly.

Correct.  But I can sympathize with why [noseshimself](https://github.com/noseshimself) might see this as a potential cause of problems for novice users who don't understand the complexity of it. Such is the danger of having advanced features.

**Comment by noseshimself** [comment]
> As with my DHCP servers, a basic ping check to make sure the IP is not responding before it hands it out should avoid two devices being given it at the same time.

The usual way this is done is someone returning to his office and putting his mobile device into some kind of dock with a wired network  connection. At least Windows is not turning off the running network interface if a new one is brought up. I've seen this happening in larger networks before and the usual fix was using different network address ranges (and VLANs) for the interfaces.

**Comment by ackstorm23** [comment]
> > As with my DHCP servers, a basic ping check to make sure the IP is not responding before it hands it out should avoid two devices being given it at the same time.
> 
> The usual way this is done is someone returning to his office and putting his mobile device into some kind of dock with a wired network connection. At least Windows is not turning off the running network interface if a new one is brought up. I've seen this happening in larger networks before and the usual fix was using different network address ranges (and VLANs) for the interfaces.

Yes, there are various alternatives that fall short of the OS-agnostic feature I am requesting. 

1.  Very specific to the operating system, and thus only work in very specific scenarios.  For example IoT devices like IPCAMs often switch between wireless and wired, but generally do not have support for sharing the MAC, IP, or allowing for bonding tricks to work around this.  
2.  Requires the device to have a different IP address for each device, which can also be undesirable in various scenarios.
